src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -246,27 +256,28 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
                         {
                             moreResults = true;
 
-                            // At this point we are at the last row.
-                            // if we have more columns, it means sort expressions were added.
-                            if (reader.FieldCount > 10)
-                            {
-                                sortValue = reader.GetValue(SortValueColumnName) as DateTime?;
-                            }
-
                             continue;
                         }
 
+                        string rawResource;
+                        using (rawResourceStream)
+                        {
+                            rawResource = await CompressedRawResourceConverter.ReadCompressedRawResource(rawResourceStream);
+                        }
```

Comment:
curious, why it's been moved?

Reply:
Calling the `reader.GetValue` on the sort column prevented us from reading from the `rawResourceStream` (that is returned from the earlier `reader.ReadRow()` call) later on. Hence I switched the ordering to read the stream first and then call the reader again.